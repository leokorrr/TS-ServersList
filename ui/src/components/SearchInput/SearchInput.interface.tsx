export default interface SeachInputInterface {
    onSearch: Function
}