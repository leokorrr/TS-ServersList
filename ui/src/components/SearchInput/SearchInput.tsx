import React from 'react'
import SearchInputInterface from './SearchInput.interface'
import searchIcon from '../../images/search.svg'

const SearchInput: React.FC<SearchInputInterface> = ({ onSearch }) => {
    const handleChange = (e: any) => {
        const { value } = e.target
        onSearch(value)
    } 
    return (
        <label htmlFor="search" className="search-input">
            <img src={searchIcon} alt="search"/>
            <input id="search" className="search-input__input-field" onChange={handleChange} placeholder="Search" type="text"/>
        </label>
    )
} 

export default SearchInput