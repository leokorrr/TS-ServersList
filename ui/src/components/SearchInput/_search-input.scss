.search-input {
    display: flex;
    align-items: center;
    position: relative;
    
    &__input-field {
        border: 0.2rem solid $color-search-border;
        border-radius: 5rem;
        font-size: 1.4rem;
        background-color: transparent;
        width: 26.3rem;
        padding: 0.9rem;
        padding-left: 4rem;

        &:focus {
            outline: none;
        }
    }

    img {
        left: 1rem;
        position: absolute;
    }
}