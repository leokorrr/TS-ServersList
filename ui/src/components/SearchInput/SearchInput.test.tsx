import React from 'react'
import SearchInput from './SearchInput'
import { shallow, mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

const onSearch = () => {}
configure({ adapter: new Adapter() })
it("SearchInput rendered without crashing", () => {
    shallow(<SearchInput onSearch={onSearch}/>);
})
it("should render initial layout", () => {
    const component = shallow(<SearchInput onSearch={onSearch} />);
    expect(component.getElements()).toMatchSnapshot();
})