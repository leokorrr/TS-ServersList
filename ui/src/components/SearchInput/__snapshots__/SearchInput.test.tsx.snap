// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render initial layout 1`] = `
Array [
  <label
    className="search-input"
    htmlFor="search"
  >
    <img
      alt="search"
      src="search.svg"
    />
    <input
      className="search-input__input-field"
      id="search"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
  </label>,
]
`;
